### DISCUSSION TITLE
Problems and update for SMSZB-120

### DISCUSSION THREAD

**Discussion by TheDrent** [Koenkk/zigbee2mqtt#13594] [Devices]
I am new with Zigbee2MQTT. I have a few devices which I use in combination with HomeAssistant. I recently bought three SMSZB-120 of Frient. I noticed that they show up as Develco products which seems to be correct. I have a few problems with it and I also have the latest firmware files (4.0.5) from the 4th of May 2022. But I'm not sure if that is the firmware which is in my devices, when starting OTA it gives the message that my devices have newer firmware then known, but the actual firmware version is not shown (which would be nice if that would be implemented). The date is the same, which suggests I have the same version.

The first question I have is how to get the latest firmware known in your system, so other users are able to update their devices. I also have the change notes.

The second question I have is that it looks like the changes made to the ZigBee-herdsman-converter of the 12th September last year don't seem valid for my device. 
When the device is configuring I get the following failure:

"Failed to configure 'Rookmelder zolder', attempt 4 (Error: Bind 0x0015bc00310151fc/35 genBasic from '0x00212effff07ddf0/1' failed (Error: status: 140) at DeconzAdapter.bind (/app/node_modules/zigbee-herdsman/src/adapter/deconz/adapter/deconzAdapter.ts:764:19) at Endpoint.bind (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:567:13) at Object.bind (/app/node_modules/zigbee-herdsman-converters/lib/reporting.js:40:9) at Object.configure (/app/node_modules/zigbee-herdsman-converters/devices/develco.js:168:13) at Configure.configure (/app/lib/extension/configure.ts:115:13) at Configure.onMQTTMessage (/app/lib/extension/configure.ts:55:21))"

Also not all data is valid, which seems to be related to the changes of the 12th of September last year, but as I’m new, I’m not sure if this is a correct conclusion:
{
    "battery": 100,
    "battery_low": false,
    "linkquality": 255,
    "restore_reports": true,
    "smoke": false,
    "supervision_reports": true,
    "test": false,
    "update": {
        "state": "idle"
    },
    "update_available": false,
    "voltage": 3000,
    "alarm": null,
    "fault": null,
    "max_duration": null,
    "reliability": null,
    "temperature": null
}

Any help is welcome.

**Comment by RobLeighton67** [comment]
Yeah, just to confirm - I successfully did an automatic OTA a couple of days ago on an older Develco SMSZB-120 from 3.x.x to 4.0.2 from the server.
It's working fine.
Z2M version was 1.26.0 with Sonoff P Dongle.

4.0.5 firmware clearly newer of course!

**Reply by SjonnyC** [reply]
Hmm, just updated to 1.27 and still have the same problems. Would it be the dongle causing the problems? I have a Conbee II stick. How can I hunt down where the problem is coming from?

And where do I leave the updated firmware? I haven't posted firmware or code before, so any help is appreciated.

**Reply by RobLeighton67** [reply]
Is it possible that the OTA server doesn't have a later version of firmware?

**Reply by SjonnyC** [reply]
I don't know which version is on the OTA server. These are the messages I get when I try to update:

warning 2022-09-02 19:31:04Firmware on '0x0015bc00310151fc' is newer than latest firmware online.
info 2022-09-02 19:31:04No update available for 'Rookmelder zolder'
info 2022-09-02 19:31:04MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"No update available for 'Rookmelder zolder'","meta":{"device":"Rookmelder zolder","status":"not_available"},"type":"ota_update"}'
info 2022-09-02 19:31:04MQTT publish: topic 'zigbee2mqtt/Rookmelder zolder', payload '{"alarm":null,"battery":100,"battery_low":false,"fault":null,"linkquality":255,"max_duration":null,"reliability":null,"restore_reports":true,"smoke":false,"supervision_reports":true,"temperature":null,"test":false,"update":{"state":"idle"},"update_available":false,"voltage":3000}'

I received the latest firmware with release notes, but I don't know how to get them on the OTA-server. But that would also not solve my problem, I guess.

As you can also see, some of the info received is "null".
